pkgs/games/redeclipse/default.nix

```
@@ -0,0 +1,53 @@
+{ stdenv, fetchFromGitHub, fetchurl, fetchpatch
+, curl, ed, pkgconfig, zlib, libX11
+, SDL2, SDL2_image, SDL2_mixer
+}:
+
+stdenv.mkDerivation rec {
+  pname = "redeclipse";
+  version = "1.6.0";
+
+  src = fetchurl {
+    url = "https://github.com/red-eclipse/base/releases/download/v1.6.0/redeclipse_1.6.0_nix.tar.bz2";
+    sha256 = "0j98zk7nivdsap4y50dlqnql17hdila1ikvps6vicwaqb3l4gaa8";
+  };
+
+  buildInputs = [
+    curl ed pkgconfig libX11 zlib
+    SDL2 SDL2_image SDL2_mixer
+  ];
+
+  makeFlags = [ "-C src/" "prefix=$(out)" ];
+
+  patches = [
+    # "remove gamma name hack" - Can't find `____gammaf128_r_finite` otherwise
+    # Is likely to be included in next release
+    (fetchpatch {
+       url = "https://github.com/red-eclipse/base/commit/b16b4963c1ad81bb9ef784bc4913a4c8ab5f1bb4.diff";
+       sha256 = "099ci77c67pyznjqrfsr14qnfc5fg7vccl6mldxcafnwiqw0wk3m";
```

`fetchpatch` produces a cleaned up patch so the hash is different from the one obtained from `fetchurl`:

```suggestion
       sha256 = "1bm07qrq60bbmbf5k9255qq115mcyfphfy2f7xl1yx40mb9ns65p";
```

---

Weird. Should that invalidate the nix-build? I did one before pushing changes and it completed normally.

---

No. Since `fetch*` functions produce fixed output derivations, the hash is the only thing that determines whether the derivation will be rebuild.